src/frontend/src/eth/services/send.services.ts

```diff
@@ -264,6 +264,14 @@ export const send = async ({
 
 	const nonce = await getTransactionCount(from);
 
+	// TODO: We may need to add an approve transaction that resets the approved amount to zero, in case of conversion from ERC20.
```

Comment:
Instead of resetting, couldn't we potentially know if the spender is already entitled to spend the amount and skip the approval if it's the case?

I guess this has been discussed with cross-chain team?

Reply:
yes, correct, we could query the allowance endpoint of the ERC20 to check. For example:

### Case 1:
- query allowance
- no amount approved --> ask for approval

### Case 2:
- query allowance
- old amount approved --> reset amount
- ask for new approval

That what you mean?

Reply:
No, I mean:

# Case 3:

- query allowance
- is allowance still valid?
- yes, ask for new approval
- no, request allowance, ask for new approval

Would that be possible ? I really don't know, just a question.